From dc0928737b7796f5d3faedfafe959fd67ed1cdb8 Mon Sep 17 00:00:00 2001
From: Scott Murray <scotmurr@codeaurora.org>
Date: Tue, 21 Apr 2021 13:14:21 -0500
Subject: [PATCH] Remove qt-qrcode usage

Remove qt-qrcode usage to avoid needing it.  This is required
since it is GPLv3 licensed.  The QR code functionality is
specific to the Alexa demo feature and thus not absolutely
required.

Upstream-Status: Inappropriate

Signed-off-by: Scott Murray <scotmurr@codeaurora.org>

diff --git a/app/app.pro b/app/app.pro
index 2ff2519..9656da0 100644
--- a/app/app.pro
+++ b/app/app.pro
@@ -4,7 +4,7 @@ QT = quickcontrols2 websockets
 SOURCES = main.cpp
 
 CONFIG += link_pkgconfig
-PKGCONFIG += qtappfw-network qtappfw-bt qtappfw-voice qtqrcode-quickitem
+PKGCONFIG += qtappfw-network qtappfw-bt qtappfw-voice
 
 CONFIG(release, debug|release) {
     QMAKE_POST_LINK = $(STRIP) --strip-unneeded $(TARGET)
diff --git a/app/main.cpp b/app/main.cpp
index 40d0d55..a48eb28 100644
--- a/app/main.cpp
+++ b/app/main.cpp
@@ -24,7 +24,6 @@
 #include <QtQml/QQmlContext>
 #include <QtQuickControls2/QQuickStyle>
 #include <QQuickWindow>
-#include <QtQrCodeQuickItem.hpp>
 #include <bluetooth.h>
 #include <network.h>
 #include <voice.h>
@@ -93,7 +92,6 @@ int main(int argc, char *argv[])
         qWarning() << aglversion.errorString();
     }
 
-    QtQrCodeQuickItem::registerQmlTypes();
     engine.rootContext()->setContextProperty("bluetooth", new Bluetooth(bindingAddressWS, context));
     engine.rootContext()->setContextProperty("voice", new Voice(bindingAddressWS, context));
 
diff --git a/app/voice/ConfigDialog.qml b/app/voice/ConfigDialog.qml
index 6ab7dea..9b9c4a8 100644
--- a/app/voice/ConfigDialog.qml
+++ b/app/voice/ConfigDialog.qml
@@ -18,7 +18,6 @@
 import QtQuick 2.11
 import QtQuick.Controls 2.4
 import QtQuick.Layouts 1.3
-import QtQrCode.Component 1.0
 import AGL.Demo.Controls 1.0
 
 Dialog {
@@ -214,18 +213,6 @@ Dialog {
                 Layout.topMargin: 8
                 Layout.bottomMargin: 8
 
-                QtQrCode {
-                    /*
-                     * FIXME: Generated URL is currently Alexa-specific, work around for
-                     *        now by checking agent name and only displaying for Alexa.
-                     */
-                    data: "%1?cbl-code=%2".arg(root.thisAgentLoginUrl).arg(root.thisAgentLoginCode)
-                    visible: root.thisAgentName === "Alexa" && root.thisAgentAuthState !== "REFRESHED" && root.tokenValid
-                    background: 'white'
-                    margin: 16
-                    Layout.fillWidth: true
-                    Layout.fillHeight: true
-                }
             }
             Label {
                 id: instructions
